11_1_scalpel.js

```diff
@@ -0,0 +1,367 @@
+        request(nest, neighbor, 'connections', {
+            name,
+            neighbors: nest.state.connections.get(name),
+        });
+    }
+}
+
+everywhere(nest => {
+    nest.state.connections = new Map();
+    nest.state.connections.set(nest.name, nest.neighbors);
+    broadcastConnections(nest, nest.name);
+});
+
+function findRoute(from, to, connections) {
+    let work = [{ at: from, via: null }];
+    for (let i = 0; i < work.length; i++) {
+        let { at, via } = work[i];
+        for (let next of connections.get(at) || []) {
+            if (next == to) return via;
+            if (!work.some(w => w.at == next)) {
+                work.push({ at: next, via: via || next });
+            }
+        }
+    }
+    return null;
+}
+
+function routeRequest(nest, target, type, content) {
+    if (nest.neighbors.includes(target)) {
+        return request(nest, target, type, content);
+    } else {
+        let via = findRoute(nest.name, target, nest.state.connections);
+        if (!via) throw new Error(`No route to ${target}`);
+        return request(nest, via, 'route', { target, type, content });
+    }
+}
+
+requestType('route', (nest, { target, type, content }) => {
+    return routeRequest(nest, target, type, content);
+});
+
+requestType('storage', (nest, name) => storage(nest, name));
+
+function findInStorage(nest, name) {
+    return storage(nest, name).then(found => {
+        if (found != null) return found;
+        else return findInRemoteStorage(nest, name);
+    });
+}
+
+function network(nest) {
+    return Array.from(nest.state.connections.keys());
+}
+
+function findInRemoteStorage(nest, name) {
+    let sources = network(nest).filter(n => n != nest.name);
+    function next() {
+        if (sources.length == 0) {
+            return Promise.reject(new Error('Not found'));
+        } else {
+            let source = sources[Math.floor(Math.random() * sources.length)];
+            sources = sources.filter(n => n != source);
+            return routeRequest(nest, source, 'storage', name).then(
+                value => (value != null ? value : next()),
+                next,
+            );
+        }
+    }
+    return next();
+}
+
+var Group = class Group {
+    constructor() {
+        this.members = [];
+    }
+    add(m) {
+        this.members.add(m);
+    }
+};
+
+function anyStorage(nest, source, name) {
+    if (source == nest.name) return storage(nest, name);
+    else return routeRequest(nest, source, 'storage', name);
+}
+
+async function chicks(nest, year) {
+    let list = '';
+    await Promise.all(
+        network(nest).map(async name => {
+            list += `${name}: ${await anyStorage(nest, name, `chicks in ${year}`)}\n`;
+        }),
+    );
+    return list;
+}
+
+//Solution
+
+//async function
+async function locateScalpel(nest, current) {
+    const next = await anyStorage(nest, current, 'scalpel');
+    if (next === current) return current;
+    else {
```

Comment:
You don't need that `else` statement. If you have a `return` in the previous `if`, then you can omit `else` and just put your code there - it will be triggered only if the previous statement won't pass.